rotkehlchen/chain/ethereum/modules/aave/v2/decoder.py

```diff
@@ -25,3 +47,101 @@ def __init__(
             pool_address=POOL_ADDRESS,
             eth_gateways=ETH_GATEWAYS,
         )
+
+    def _decode_staking_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode aave v2 staking unstaking events"""
+        decode_map = {
+            STAKED_AAVE: self._decode_stake,
+            REDEEM_AAVE: self._decode_unstake,
+        }
```

Comment:
don't create a map. You can do an if/else and then assign a variable `decoding_logic = self._decode...`. Creating a map is more expensive